net.go

```diff
@@ -203,6 +203,33 @@ func (n *Net) Walk(r *net.IPNet, handle func(*net.IPNet, interface{}) bool) {
 	})
 }
 
+// WalkPrefix interates routes that have a given prefix.
+// handle is called with arguments route and value (if handle returns `false`, the iteration is aborted)
+func (n *Net) WalkPrefix(r *net.IPNet, handle func(*net.IPNet, interface{}) bool) {
+	var prefix []byte
+	var div int
+	var bit uint
+	if r != nil {
+		if ip, _, err := netValidateIPNet(r); err == nil {
+			prefix = netIPNetToKey(ip, r.Mask)
+			mask := prefix[len(prefix)-1]
+			div = int(mask >> 3)
+			if mod := uint(mask & 0x07); mod != 0 {
+				bit = 8 - mod
+			}
+		}
+	}
+	wrapper := func(key []byte, value interface{}) bool {
+		if bit != 0 {
+			if prefix[div]>>bit != key[div]>>bit {
+				return true
```

Comment:
```diff
-				return true
+				return false
```
If true, I think will unnecessary repetition occurs in `Allprefixed`.

Reply:
Ah, fixed. Thanks!